libs/datamodel/core/src/transform/ast_to_dml/generator_loader.rs

```diff
@@ -71,10 +96,18 @@ impl GeneratorLoader {
         };
 
         if !preview_features.is_empty() {
-            if let Err(err) =
-                validate_preview_features(preview_features.clone(), span, Vec::from(GENERATOR_PREVIEW_FEATURES))
-            {
-                return Err(err);
+            let mut result = validate_preview_features(
+                preview_features.clone(),
+                span,
+                Vec::from(GENERATOR_PREVIEW_FEATURES),
+                Vec::from(DEPRECATED_GENERATOR_PREVIEW_FEATURES),
+            );
+            if result.has_errors() {
+                diagnostics.append(&mut result);
+                return Err(diagnostics);
+            }
+            if result.has_warnings() {
+                diagnostics.append_warning_vec(result.warnings)
```

Comment:
Shouldn't the warnings be appended before the error case? This way you don't see warnings in the case of an error. But maybe this can't happen at all or you are not trusting the warnings if there are errors 🤷

Reply:
Good point, they should be before I think.